### PR TITLE
Consider secure option in service discovery configuration

### LINKED ISSUE #25421: Stork extension ignores `secure` property
### Describe the bug

When a Stork service is configured with the `secure` property, the reactive REST client ignores this and instead uses HTTP or HTTPS based on whether the REST client's URL used the `stork` or `storks` scheme. 

### Expected behavior

At runtime the `ServiceInstance#isSecure()` should be used to determine whether to use HTTP or HTTPS.

### Actual behavior

The client's URI scheme (`stork` or `storks`) controls the protocol.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.9.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ejba**
Fixes #25421

### CODE REVIEW

**Comment by sberyozkin** [comment]
Is a flag alone enough to have HTTPS connection working for Stork ? Should trust certificates be configured or is the REST client configuration reused for Stork ?

**Comment by ejba** [comment]
@sberyozkin I assumed it is after reading the issue's description, but I could be wrong. I am still working on an IT to prove it.

**Comment by ejba** [comment]
> Is a flag alone enough to have HTTPS connection working for Stork ? Should trust certificates be configured or is the REST client configuration reused for Stork ?

`Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target`  So, I guess them need to be configured, but I don't know how to do it to be honest @sberyozkin

File: extensions/smallrye-stork/runtime/src/main/java/io/quarkus/stork/StorkConfigUtil.java [hunk 1]
```diff
@@ -15,10 +15,13 @@ public static List<ServiceConfig> toStorkServiceConfig(StorkConfiguration storkC
         SimpleServiceConfig.Builder builder = new SimpleServiceConfig.Builder();
         for (String serviceName : servicesConfigs) {
             builder.setServiceName(serviceName);
+
             ServiceConfiguration serviceConfiguration = storkConfiguration.serviceConfiguration.get(serviceName);
             SimpleServiceConfig.SimpleServiceDiscoveryConfig storkServiceDiscoveryConfig = new SimpleServiceConfig.SimpleServiceDiscoveryConfig(
                     serviceConfiguration.serviceDiscovery.type, serviceConfiguration.serviceDiscovery.params);
             builder.setServiceDiscovery(storkServiceDiscoveryConfig);
+            builder.setSecure(serviceConfiguration.serviceDiscovery.secure);
```
Comment (michalszynkiewicz) [inline extensions/smallrye-stork/runtime/src/main/java/io/quarkus/stork/StorkConfigUtil.java:23]: I think this won't work with Stork 1.1.2, I'll update Quarkus to it this/next week and we'll see when we rebase the PR

**Comment by michalszynkiewicz** [comment]
REST client configuration should be reused, there should not be anything specific around cert configuration for Stork

File: integration-tests/rest-client-reactive-stork/src/test/java/io/quarkus/it/rest/reactive/stork/SlowWiremockServer.java [hunk 2]
```diff
@@ -12,15 +12,20 @@ public class SlowWiremockServer extends WiremockBase {
     static final String SLOW_RESPONSE = "hello, I'm a slow server";
 
     @Override
-    int port() {
+    int httpPort() {
         return 8767;
     }
 
+    @Override
+    int httpsPort() {
```
Comment (michalszynkiewicz) [inline integration-tests/rest-client-reactive-stork/src/test/java/io/quarkus/it/rest/reactive/stork/SlowWiremockServer.java:20]: do I get it correctly that the httpsPort is not used anywhere?

**Comment by ejba** [comment]
> REST client configuration should be reused, there should not be anything specific around cert configuration for Stork

the only thing it's done here is in testing to accept self-signed cert owned by wiremock.

**Review by michalszynkiewicz — APPROVED** [review]

**Comment by michalszynkiewicz** [comment]
thank you @ejba !

**Pull request merged**